mmdet/models/detectors/single_stage_instance_seg.py

```diff
@@ -0,0 +1,168 @@
+            gt_bboxes (list[Tensor]): Each item are the truth boxes for each
+                image in [tl_x, tl_y, br_x, br_y] format.
+            gt_labels (list[Tensor]): Class indices corresponding to each box
+            gt_bboxes_ignore (None | list[Tensor]): Specify which bounding
+                boxes can be ignored when computing the loss.
+            gt_masks (list[:obj:`BitmapMasks`]) : The segmentation
+                masks for each box.
+
+        Returns:
+            dict[str, Tensor]: A dictionary of loss components.
+        """
+
+        gt_masks = [
+            gt_mask.to_tensor(dtype=torch.bool, device=img.device)
+            for gt_mask in gt_masks
+        ]
+        x = self.extract_feat(img)
+        losses = dict()
+
+        # CondInst, yolact
+        if self.bbox_head:
+            # bbox_head_results is a tuple
+            bbox_head_preds = self.bbox_head(x)
+            # positive_infos is a obj:`DetectionResults`
+            # It contains the information about the positive samples
+            # CondInst, Yolact
+            det_losses, positive_infos = self.bbox_head.loss(
+                *bbox_head_preds,
+                gt_bboxes=gt_bboxes,
+                gt_labels=gt_labels,
+                img_metas=img_metas,
+                gt_bboxes_ignore=gt_bboxes_ignore)
+            losses.update(det_losses)
+        else:
+            positive_infos = None
+
+        mask_head_inputs = (x, gt_labels, gt_masks, img_metas)
+
+        # when no positive_infos add gt bbox
+        mask_loss = self.mask_head.forward_train(
+            *mask_head_inputs,
+            positive_infos=positive_infos,
+            gt_bboxes=gt_bboxes,
+            gt_bboxes_ignore=gt_bboxes_ignore)
+        # avoid loss override
+        assert not set(mask_loss.keys()) & set(losses.keys())
+
+        losses.update(mask_loss)
+        return losses
+
+    def simple_test(self, img, img_metas, rescale=False):
+        """Test function without test-time augmentation.
+
+        Args:
+            img (torch.Tensor): Images with shape (B, C, H, W).
+            img_metas (list[dict]): List of image information.
+            rescale (bool, optional): Whether to rescale the results.
+                Defaults to False.
+
+        Returns:
+            list[:obj:`DetectionResults`]: Processed results of multiple
+            images.Each :obj:`DetectionResults` usually contains
+            following keys.
+
+                - scores (Tensor): Classification scores, has shape
+                  (num_instance,)
+                - labels (Tensor): Has shape (num_instances,).
+                - masks (Tensor): Processed mask results, has
+                  shape (num_instances, h, w).
+        """
+        feat = self.extract_feat(img)
+        if self.bbox_head:
+            # det_results is a obj:`DetectionResults`
+            outs = self.bbox_head(feat)
+            results_list = self.bbox_head.get_bboxes(
```

Comment:
The results_list is a list or a obj of DetectionResults?

Reply:
yes

Reply:
Seems renaming it to `get_results` would be better.